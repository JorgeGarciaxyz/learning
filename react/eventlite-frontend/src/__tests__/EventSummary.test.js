import { render, screen } from '@testing-library/react';
import EventSummary from '../components/EventSummary';
import { BrowserRouter as Router } from 'react-router-dom'
import TestRenderer from 'react-test-renderer';

const event = {
  title: "My title",
  start_datetime: "10-10-2022",
  location: "my house"
}

it('renders event title', () => {
  render(
  <Router>
    <EventSummary event={event} />
  </Router>
  )

  const titleElement = screen.getByText(event.title);
  expect(titleElement).toBeInTheDocument();
  // If there's an image
  // const image = getByRole('img', {src: 'event.png'})
})

it('should render an event summary card', () => {
  //render(<EventSummary />)
  render(
    <Router>
      <EventSummary event={event} />
    </Router>
    )
  expect(document.body.innerHTML).toMatchSnapshot()
})

// Not working IDK why
it('should render an event summary card renderer', () => {
  //render(<EventSummary />)
  const tree = TestRenderer.create(<EventSummary event={event} />).toJSON();

  expect(tree).toMatchSnapshot();
})
