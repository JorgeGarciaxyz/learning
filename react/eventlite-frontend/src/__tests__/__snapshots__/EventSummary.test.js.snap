// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an event summary card 1`] = `"<div><div class=\\"event\\"><a style=\\"text-decoration: none;\\" href=\\"/events/undefined\\"><h2 class=\\"event-title\\">My title</h2></a></div></div>"`;
